src/main/java/org/openrewrite/java/migrate/io/RemoveFinalizeFromFileStream.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate.io;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.List;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveFinalizeFromFileStream extends Recipe {
+
+    private static final String JAVA_IO_FILEINPUTSTREAM = "java.io.FileInputStream";
+    private static final String JAVA_IO_FILEOUTPUTSTREAM = "java.io.FileOutputStream";
+
+    private static final MethodMatcher METHOD_MATCHER = new MethodMatcher("java.lang.Object finalize()");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace invocations of deprecated invocations from FileInputStream, FileOutputStream";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove invocations of finalize() deprecated invocations from FileInputStream, FileOutputStream.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+
+        return Preconditions.check(
+                Preconditions.or(
+                        new UsesType<>(JAVA_IO_FILEINPUTSTREAM, false),
+                        new UsesType<>(JAVA_IO_FILEOUTPUTSTREAM, false)),
```

Comment:
Reading JDK-8212050 I see:
> The java.lang.ref.Cleaner has been implemented since JDK 9 as the primary mechanism to close FileDescriptors no longer reachable from FileInputStream and FileOutputStreams.
Previously, finalize would call the close method. The recommendation has been to explicitly call close or use try-with-resources to close files.
With this change the FileInputStream or FileOutputStream close method will no longer be called when a FileInputstream or FileOutputStream is found to be unreachable.

I'd _think_ it's then best to already migrate projects on Java 9+, but not any projects using Java 8 or below. If you agree we can add another precondition to limit recipe execution.
```suggestion
                Preconditions.and(
                        new UsesJavaVersion<>(9),
                        Preconditions.or(
                                new UsesType<>(JAVA_IO_FILEINPUTSTREAM, false),
                                new UsesType<>(JAVA_IO_FILEOUTPUTSTREAM, false))),
```